code/datums/beam.dm

```diff
@@ -17,6 +17,8 @@
 	var/recalculating = FALSE
 
 /datum/beam/New(beam_origin,beam_target,beam_icon='icons/effects/beam.dmi',beam_icon_state="b_beam",time=50,maxdistance=10,btype = /obj/effect/ebeam,beam_sleep_time=3)
+	if(time >= INFINITY)
```

Comment:
Adding the timer despite this is also probably a bad idea. I'd have done this at the bottom:
```dm
	if (time >= INFINITY)
		qdel(src)
	else
		addtimer(CALLBACK(src, .proc/End), time)
```

Alternatively. Implement a `CRASH` at the top of this proc, and move the actual sanity check into the `/atom/proc/Beam` proc:
```dm
/atom/proc/Beam(atom/BeamTarget,icon_state="b_beam",icon='icons/effects/beam.dmi',time=50, maxdistance=10,beam_type=/obj/effect/ebeam,beam_sleep_time = 3)
	if (time < INFINITY)
		var/datum/beam/newbeam = new(src,BeamTarget,icon,icon_state,time,maxdistance,beam_type,beam_sleep_time)
		INVOKE_ASYNC(newbeam, /datum/beam/.proc/Start)
		return newbeam
	else
		crash_with("Attempted to beam() with an infinite time.")
		return null
```

Reply:
I think just adding `return` after `qdel(src)` is sufficient. As I was trying to avoid beam calculations and drawing to spend less CPU on invalid beams.